### PR TITLE
Worker cleanup

### LINKED ISSUE #21: Remove unused options like sdsflags
IIRC `sdsflags` was a stop gap measure for when we didn't have the `make` based builds. Is it used anywhere anymore @sampsyo @sa2257? If not, we should remove it.

### PR DESCRIPTION
**Pull request by rachitnigam**
PR does a few things
- Separate out SDSoC and F1 specific stages into different functions. The only place where the toolchain check happens is in worker.py. Other stages should not be aware of it.
- Separate out common stage functionality.
- Remove unused code path for automatically building kernels using SDSoC. `make` based SDSoC path still works. (Fixes #21)

Deployment tests:
- [X] F1
- [ ] Gorgonzola

### CODE REVIEW

File: polyphemus/worker.py [hunk 1]
```diff
@@ -181,7 +48,6 @@ def run(self):
         while True:
             self.func(self.db, self.config)
 
-
```
Comment (sampsyo) [inline polyphemus/worker.py]: Two blanks between top-level functions are intentional ([à la PEP 8](https://www.python.org/dev/peps/pep-0008/#blank-lines)).

**Comment by rachitnigam** [comment]
PR does a few things
- Separate out SDSoC and F1 specific stages into different functions. The only place where the toolchain check happens is in worker.py. Other stages should not be aware of it.
- Separate out common stage functionality.
- Remove unused code path for automatically building kernels using SDSoC. `make` based SDSoC path still works.

Deployment tests:
- [ ] F1: Not done
- [ ] Gorgonzola: Not done.

**Comment by rachitnigam** [comment]
Works fine with F1. @sa2257 whenever you redeploy Gorgonzola, let me know if things break.

Merging changes by EOD.

File: Makefile [hunk 2]
```diff
@@ -8,5 +8,8 @@ dev:
 serve:
 	pipenv run gunicorn --bind 0.0.0.0:$(PORT) polyphemus.server:app
 
+worker:
+	pipenv run python -m polyphemus.workproc
```
Comment (sampsyo) [inline Makefile]: Good call. :+1:

**Pull request merged**